doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,715 @@
+### [DRAFT, UNDER DEVELOPMENT]
+
+- [Overview](#overview)
+    - [Scope](#scope)
+    - [Related DUT CLI commands](#related-dut-cli-commands)
+    - [SAI APIs](#sai-apis)
+- [General test flow](#general-test-flow)
+- [Run test](#run-test)
+- [Test cases](#test-cases)
+    - [Test case #1](#test-case-1)
+    - [Test case #2](#test-case-2)
+    - [Test case #3](#test-case-3)
+    - [Test case #4](#test-case-4)
+    - [Test case #5](#test-case-5)
+    - [Test case #6](#test-case-6)
+    - [Test case #7](#test-case-7)
+    - [Test case #8](#test-case-8)
+    - [Test case #9](#test-case-9)
+    - [Test case #10](#test-case-10)
+    - [Test case #11](#test-case-11)
+    - [Test case #12](#test-case-12)
+    - [Test case #13](#test-case-13)
+    - [Test case #14](#test-case-14)
+    - [Test case #15](#test-case-15)
+    - [Test case #16](#test-case-16)
+
+#### Overview
+The purpose is to test "RX_DRP" counter got from show command "show interfaces counters" triggers on receiving specific packets by DUT.
+The "RX_DRP" counter counts all discard events. This counter counts concurrently with other discard counters.
+The test assumes all necessary configuration are already pre-configured on the SONIC switch before test runs.
+Destination IP address of the injected packet must be routable to ensure packet should not be routed but dropped.
+
+#### Scope
+The purpose of test cases is to verify that "RX_DRP" counter triggers on SONIC system, making sure that specific traffic drops correctly, according to sent packet and configured packet discards.
+Supported topologies:
+```
+t0
+t1
+t1-lag
+ptf32
+```
+
+#### Discard groups covered by test case
+| Test case ID| Group type|
+|-------------|---------|
+|Test case #1 | Ethernet|
+|Test case #2 | Ethernet|
+|Test case #3 | Ethernet|
+|Test case #4 | Ethernet|
+|Test case #5 | IP|
+|Test case #6 | IP|
+|Test case #7 | IP|
+|Test case #8 | IP|
+|Test case #9 | IP|
+|Test case #10 | IP|
+|Test case #11 | IP|
+|Test case #12 | IP|
+|Test case #13 | IP|
+|Test case #14 | IP|
+|Test case #15 | IP|
+|Test case #16 | IP|
+|Test case #17 | IP|
+|Test case #18 | IP|
+
+
+#### Related DUT CLI commands
+| **Command**                                                      | **Comment** |
+|------------------------------------------------------------------|-------------|
+| show interfaces counters              | Check ```RX_DRP```                     |
+| counterpoll rif enable                | Enable RIF counters                    |
+| show interface counters rif           | Show RIF counters                      |
+| aclshow -a                            | Check ```PACKETS COUNT```              |
+
+#### SAI APIs
```

Comment:
This section is called ```SAI APIs``` but only SAI attributes are described and no API description. Please align.
If other SAI attributes are going to be used please add them as well.

Reply:
Fixed.